src/main/java/edu/cornell/kfs/sys/batch/service/impl/CuSchedulerServiceImpl.java

```diff
@@ -340,6 +341,49 @@ public void scheduleExceptionMessageJob(PersistedMessage message, String descrip
             }
         }
     }
+    
+    @Override
+    public void scheduleDelayedAsyncCallJob(PersistedMessage message, String description) {
+        JobDetail jobDetail = createDelayedAsyncCallJobDetail(message, description);
+        CuJobEntry jobEntry = new CuJobEntry(delayedAsyncCallJob, jobDetail);
+        JobKey jobKey = jobDetail.getKey();
+        boolean jobWasScheduled = false;
+        
+        if (jobs.putIfAbsent(jobKey, jobEntry) != null) {
+            throw new IllegalStateException("Job with name '" + jobKey.getName()
+                    + "' was already running; this should NEVER happen");
+        }
+        
+        try {
+            Date queueDate = new Date(message.getQueueDate().getTime());
+            runJob(jobEntry, Map.of(), queueDate);
+            jobWasScheduled = true;
+        } catch (RuntimeException e) {
+            jobWasScheduled = false;
+            LOG.error("scheduleDelayedAsyncCallJob, Could not successfully schedule job", e);
+            throw e;
+        } finally {
+            if (!jobWasScheduled) {
+                jobs.remove(jobKey);
+            }
+        }
+    }
+    
+    private JobDetail createDelayedAsyncCallJobDetail(PersistedMessage message, String description) {
+        String jobName = "Delayed_Asynchronous_Call-" + Math.random();
+        
+        JobDataMap jobDataMap = new JobDataMap();
+        jobDataMap.put(MessageServiceExecutorJob.MESSAGE_KEY, message);
+        
+        JobBuilder jobBuilder = JobBuilder.newJob();
+        if (StringUtils.isNotBlank(description)) {
+            jobBuilder = jobBuilder.withDescription(description);
+        }
+        return jobBuilder.withIdentity(jobName, "Delayed_Asynchronous_Call")
```

Comment:
There are several places in this class where the processing has been set up to behave differently for "Exception Messaging" jobs. Many of those areas may need adjustments to also have similar behavior for these delayed-async-call jobs. You might need to modify at least the following areas:

* The getJobIfPresent() method
* The 6-arg runJob() method
* The runJobWithoutQuartz() method

Reply:
updated